eeco/utils.py

```diff
@@ -443,6 +443,117 @@ def const_rule(model, t):
         )
 
 
+def get_decomposed_var_names(utility):
+    """Get consistent variable names for decomposed consumption variables."""
+    return f"{utility}_positive", f"{utility}_negative"
+
+
+def decompose_consumption(
```

Comment:
If we have a few types of decomposition, I could see how this could be a bit tough to follow. You might consider having a function for each decomposition type (i.e. `decompose_absolute_value`, `decompose_binary`, etc.) that are called within `decompose_consumption`.